src/Orleans/Lifecycle/ILifecycleObserver.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Threading.Tasks;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Lifecycle observer used to handle start and stop notification.
+    /// </summary>
+    public interface ILifecycleObserver
+    {
+        /// <summary>
+        /// Handle start notifications
+        /// </summary>
+        /// <returns></returns>
+        Task OnStart();
+
+        /// <summary>
+        /// Handle stop notifications
+        /// </summary>
+        /// <returns></returns>
+        Task OnStop();
+    }
```

Comment:
Agreed. Let's keep it simple until needed. And I see this is also borrowing pattern from reactive programming. Just had a read on how to do testing and debugging in rx : https://msdn.microsoft.com/en-us/library/hh242967(v=vs.103).aspx. 
And Querying Observable Sequences using LINQ Operators: https://msdn.microsoft.com/en-us/library/hh242983(v=vs.103).aspx

Probably a overkill for this case. But It is a good read and maybe useful in the future for this or streaming.